gtest-parallel

```diff
@@ -281,6 +290,14 @@ elif options.format == 'filter':
 else:
   sys.exit("Unknown output format: " + options.format)
 
+if options.shard_count < 1:
+  sys.exit("Invalid number of shards: %d. Must be at least 1." %
+           options.shard_count)
+if options.shard_index < 0 or options.shard_count <= options.shard_index:
```

Comment:
Oh, right—it's the test on line 293 that's redundant, but you want a separate test in order to be able to give a good error message. How about this, though? I find this easier to read:

  if not (0 <= options.shard_index < options.shard_count):